gtsfm/evaluation/metrics_report.py

```diff
@@ -125,22 +265,110 @@ def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]
     """
     scalar_metrics = {}
     metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
-
     # Separate the scalar metrics.
     for metric_name, value in metrics_dict.items():
         if isinstance(value, dict):
             # Metrics with a dict representation must contain a summary.
             if metrics.SUMMARY_KEY not in value:
                 raise ValueError(f"Metric {metric_name} does not contain a summary.")
-            # Add a scalar metric for mean of 1D distributions.
-            scalar_metrics["mean_" + metric_name] = value[metrics.SUMMARY_KEY]["mean"]
+            # Add a scalar metric for median of 1D distributions.
+            scalar_metrics["median_" + metric_name] = value[metrics.SUMMARY_KEY]["median"]
         else:
             scalar_metrics[metric_name] = value
     table = create_table_for_scalar_metrics(scalar_metrics)
     plots_fig = create_plots_for_distributions(metrics_dict)
     return table, plots_fig
 
 
+def get_figures_for_metrics_and_compare(
+    metrics_group: GtsfmMetricsGroup, other_metrics_group: List[GtsfmMetricsGroup], pipeline_names: List[str]
+) -> Tuple[str, str]:
+    """Gets the tables and plots for individual metrics in a metrics group.
+
+    All scalar metrics are reported in the table.
+    Metrics of 1-D distributions have an entry in the table for the median,
+    and a histogram or box plot as per their property.
+
+    Args:
+        metrics_group: A GtsfmMetricsGroup for any gtsfm module (frontend, data association, bundle adjustment etc).
+        other_metrics_group: A GtsfmMetricsGroup for corresponding modules from other SfM pipelines
+        pipeline_names: A list of other SfM pipeline names
+
+    Returns:
+        A tuple of table and plotly figures as HTML code.
+    """
+    scalar_metrics = defaultdict(list)
+
+    gtsfm_metric_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
+    other_pipelines_metric_dicts = []
+    for other_pipeline_metrics_group in other_metrics_group:
+        other_pipeline_dict = other_pipeline_metrics_group.get_metrics_as_dict()[other_pipeline_metrics_group.name]
+        other_pipelines_metric_dicts.append(other_pipeline_dict)
+
+    for gtsfm_metric_name, gtsfm_metric_value in gtsfm_metric_dict.items():
+        other_pipelines_metrics = defaultdict(list)
+        for other_pipeline_metric_dict in other_pipelines_metric_dicts:
+            if gtsfm_metric_name in other_pipeline_metric_dict:
+                other_pipeline_metric_value = other_pipeline_metric_dict[gtsfm_metric_name]
+                other_pipelines_metrics[gtsfm_metric_name].append(other_pipeline_metric_value)
+            else:
+                other_pipelines_metrics[gtsfm_metric_name].append("")
+        if isinstance(gtsfm_metric_value, dict):
```

Comment:
I think there is some repeated code below that can be refactored into a reusable function.

Reply:
I refactored the code into the function `add_scalar_metric()`.